### ISSUE TITLE
torch.softmax missing

### ISSUE THREAD

**Issue by toolgood** [dotnet/TorchSharp#901]
python code
``` py
    def time_delay_agg_training(self, values, corr):
        """
        SpeedUp version of Autocorrelation (a batch-normalization style design)
        This is for the training phase.
        """
        head = values.shape[1]
        channel = values.shape[2]
        length = values.shape[3]
        # find top k
        top_k = int(self.factor * math.log(length))
        mean_value = torch.mean(torch.mean(corr, dim=1), dim=1)
        index = torch.topk(torch.mean(mean_value, dim=0), top_k, dim=-1)[1]
        weights = torch.stack([mean_value[:, index[i]] for i in range(top_k)], dim=-1)
        # update corr
        tmp_corr = torch.softmax(weights, dim=-1)  # this
        # aggregation
        tmp_values = values
        delays_agg = torch.zeros_like(values).float()
        for i in range(top_k):
            pattern = torch.roll(tmp_values, -int(index[i]), -1)
            delays_agg = delays_agg + pattern * \
                         (tmp_corr[:, i].unsqueeze(1).unsqueeze(1).unsqueeze(1).repeat(1, head, channel, length))
        return delays_agg
```

**Comment by ChengYen-Tang** [comment]
torch.nn.Softmax
https://pytorch.org/docs/stable/generated/torch.nn.Softmax.html#torch.nn.Softmax

**Comment by NiklasGustafsson** [comment]
https://github.com/dotnet/TorchSharp/blob/main/src/TorchSharp/NN/Activation/Softmax.cs

**Comment by toolgood** [comment]
code : https://github.com/thuml/Autoformer/blob/main/layers/AutoCorrelation.py 
line 40

**Comment by ChengYen-Tang** [comment]
@toolgood,

Try it.
Python:
<img width="660" alt="image" src="https://user-images.githubusercontent.com/32535494/216822900-c7fc33de-2630-4727-840d-51bb09ed7009.png">

C#:
<img width="751" alt="image" src="https://user-images.githubusercontent.com/32535494/216822967-d6ed0a03-85d5-4f32-a3ce-6e9020dcff4a.png">

**Comment by toolgood** [comment]
@ChengYen-Tang  thank you .

How to handle the method that the document does not have？

**Comment by ChengYen-Tang** [comment]
Go to the documentation to find a method for the same function?

**Comment by toolgood** [comment]
I want to build an extension library to record the method of not saving documents

**Comment by NiklasGustafsson** [comment]
I will add this undocumented function to TorchSharp. Please note that the interface is different between `torch.nn.functional.softmax()` and `torch.special.softmax()` -- the former has a default value for 'dim' (None), and the latter has no default value. I'm going with the latter for `torch.softmax()`
